File: js/src/test/Exchange/base/test.transfer.js
import testSharedMethods from './test.sharedMethods.js';
function testTransfer(exchange, skippedProperties, method, entry, requestedCode) {
    const format = {
        'info': {},
        'id': '1234',
        'timestamp': 1502962946216,
        'datetime': '2017-08-17 12:42:48.000',
        'currency': 'USDT',
        'amount': exchange.parseNumber('1.234'),
        'fromAccount': 'spot',
        'toAccount': 'swap',
        'status': 'ok',
    };
    const emptyAllowedFor = ['fromAccount', 'toAccount'];
    testSharedMethods.assertStructure(exchange, skippedProperties, method, entry, format, emptyAllowedFor);
    testSharedMethods.assertTimestampAndDatetime(exchange, skippedProperties, method, entry, exchange.milliseconds());
    testSharedMethods.assertCurrencyCode(exchange, skippedProperties, method, entry, entry['currency'], requestedCode);
    //
    testSharedMethods.assertInArray(exchange, skippedProperties, method, entry, 'status', ['ok', 'pending', 'failed']);
    testSharedMethods.assertGreaterOrEqual(exchange, skippedProperties, method, entry, 'amount', '0');
}
export default testTransfer;

File: julia/Ccxt/test/validators/test.transfer.jl
using Test
using Ccxt
function testTransfer(exchange, skippedProperties, method, entry, requestedCode)

    format = Dict{Symbol, Any}(
        Symbol("info") => Dict{Symbol, Any}(),
        Symbol("id") => "1234",
        Symbol("timestamp") => 1502962946216,
        Symbol("datetime") => "2017-08-17 12:42:48.000",
        Symbol("currency") => "USDT",
        Symbol("amount") => parseNumber(exchange, "1.234"),
        Symbol("fromAccount") => "spot",
        Symbol("toAccount") => "swap",
        Symbol("status") => "ok"
    );
    emptyAllowedFor = ["fromAccount", "toAccount"];
    assertStructure(testSharedMethods, exchange, skippedProperties, method, entry, format, emptyAllowedFor);
    assertTimestampAndDatetime(testSharedMethods, exchange, skippedProperties, method, entry, milliseconds(exchange));
    assertCurrencyCode(testSharedMethods, exchange, skippedProperties, method, entry, get(entry, Symbol("currency"), nothing), requestedCode);
    assertInArray(testSharedMethods, exchange, skippedProperties, method, entry, "status", ["ok", "pending", "failed"]);
    assertGreaterOrEqual(testSharedMethods, exchange, skippedProperties, method, entry, "amount", "0");
end
